engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala

```diff
@@ -84,24 +101,36 @@ class EngineJobExecutionActor(jobKey: BackendJobDescriptorKey,
 
   // When CheckingCallCache, the FSM always has EJEAJobDescriptorData
   when(CheckingCallCache) {
+    case Event(HashError(t), EJEAJobDescriptorData(Some(jobDescriptor), Some(bjeaProps))) =>
+      writeToMetadata(Map("Call caching read result" -> s"Hashing Error: ${t.getMessage}"))
+      recordHashError(t)
+      runJob(jobDescriptor, bjeaProps)
     case Event(CacheMiss, EJEAJobDescriptorData(Some(jobDescriptor), Some(bjeaProps))) =>
+      writeToMetadata(Map("Call caching read result" -> "Cache Miss"))
       log.info(s"Cache miss for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}")
       runJob(jobDescriptor, bjeaProps)
     case Event(CacheHit(cacheResultId), EJEAJobDescriptorData(Some(jobDescriptor), _)) =>
+      writeToMetadata(Map("Call caching read result" -> s"Cache Hit (from result ID $cacheResultId)"))
       log.info(s"Cache hit for job ${jobDescriptor.key.call.fullyQualifiedName}, index ${jobDescriptor.key.index}! Copying cache result $cacheResultId")
       lookupCachedResult(jobDescriptor, cacheResultId)
   }
 
   // When RunningJob, the FSM always has EJEAPartialCompletionData (which might be None, None)
   when(RunningJob) {
-    case Event(response: SucceededResponse, EJEAPartialCompletionData(None, Some(hashes))) if callCachingMode.writeToCache =>
+    case Event(response: SucceededResponse, EJEAPartialCompletionData(None, Some(Success(hashes)))) if effectiveCallCachingMode.writeToCache =>
       saveCacheResults(EJEASuccessfulCompletionDataWithHashes(response, hashes))
-    case Event(response: SucceededResponse, data @ EJEAPartialCompletionData(None, None)) if callCachingMode.writeToCache =>
+    case Event(response: SucceededResponse, data @ EJEAPartialCompletionData(None, None)) if effectiveCallCachingMode.writeToCache =>
       stay using data.copy(jobResult = Option(response))
     case Event(hashes: CallCacheHashes, data @ EJEAPartialCompletionData(Some(response: SucceededResponse), None)) =>
       saveCacheResults(EJEASuccessfulCompletionDataWithHashes(response, hashes))
     case Event(hashes: CallCacheHashes, data @ EJEAPartialCompletionData(None, None)) =>
-      stay using data.copy(hashes = Option(hashes))
+      stay using data.copy(hashes = Option(Success(hashes)))
+    case Event(HashError(t), data @ EJEAPartialCompletionData(Some(response: BackendJobExecutionResponse), _)) =>
+      recordHashError(t)
+      saveJobCompletionToJobStore(response)
+    case Event(HashError(t), data @ EJEAPartialCompletionData(None, _)) =>
+      recordHashError(t)
+      stay using data.copy(hashes = Some(Failure(t)))
```

Comment:
I'm on a mission to make people feel bad for calling me out on this (but I will change it): 

I'm having to move against "what I mean" in the code and move towards "arbitrary rule". It makes things harder to read and harder to understand for 0 benefit in the end result.